drivers/drv8825/src/drv8825.rs

```diff
@@ -0,0 +1,369 @@
+    pub fn from_step_dir_pins<Error>(step: Step, dir: Dir) -> Self
+    where
+        Step: OutputPin<Error = Error>,
+        Dir: OutputPin<Error = Error>,
+    {
+        Self {
+            enable: (),
+            fault: (),
+            sleep: (),
+            reset: (),
+            mode0: (),
+            mode1: (),
+            mode2: (),
+            step,
+            dir,
+        }
+    }
+}
+
+impl<Step, Dir> DRV8825<(), (), (), (), (), (), (), Step, Dir> {
+    /// Enables support for step mode control and sets the initial step mode
+    ///
+    /// Consumes this instance of `DRV8825` and returns another instance that
+    /// has support for controlling the step mode. Requires the additional pins
+    /// for doing so, namely RESET, MODE0, MODE1, and MODE2. It expects the
+    /// types that represent those pins to implement [`OutputPin`].
+    ///
+    /// This method is only available when those pins have not been provided
+    /// yet. After this method has been called once, you can use
+    /// [`DRV8825::set_step_mode`] to change the step mode again.
+    pub fn enable_mode_control<
+        Reset,
+        Mode0,
+        Mode1,
+        Mode2,
+        Clk,
+        OutputPinError,
+    >(
+        self,
+        reset: Reset,
+        mode0: Mode0,
+        mode1: Mode1,
+        mode2: Mode2,
+        step_mode: StepMode,
+        clock: &Clk,
+    ) -> Result<
+        DRV8825<(), (), (), Reset, Mode0, Mode1, Mode2, Step, Dir>,
+        ModeError<OutputPinError>,
+    >
+    where
+        Reset: OutputPin<Error = OutputPinError>,
+        Mode0: OutputPin<Error = OutputPinError>,
+        Mode1: OutputPin<Error = OutputPinError>,
+        Mode2: OutputPin<Error = OutputPinError>,
+        Clk: Clock,
+    {
+        let mut self_ = DRV8825 {
+            enable: self.enable,
+            fault: self.fault,
+            sleep: self.sleep,
+            reset,
+            mode0,
+            mode1,
+            mode2,
+            step: self.step,
+            dir: self.dir,
+        };
+
+        self_.set_step_mode(step_mode, clock)?;
+
+        Ok(self_)
+    }
+}
+
+impl<Reset, Mode0, Mode1, Mode2, Step, Dir, OutputPinError> SetStepMode
+    for DRV8825<(), (), (), Reset, Mode0, Mode1, Mode2, Step, Dir>
+where
+    Reset: OutputPin<Error = OutputPinError>,
+    Mode0: OutputPin<Error = OutputPinError>,
+    Mode1: OutputPin<Error = OutputPinError>,
+    Mode2: OutputPin<Error = OutputPinError>,
+{
+    type Error = ModeError<OutputPinError>;
+
+    /// Sets the step mode
+    ///
+    /// This method is only available, if all the pins required for setting the
+    /// step mode have been provided using [`DRV8825::enable_mode_control`].
+    fn set_step_mode<Clk: Clock>(
+        &mut self,
+        step_mode: StepMode,
+        clock: &Clk,
+    ) -> Result<(), Self::Error> {
+        // 7.6 Timing Requirements (page 7)
+        // https://www.ti.com/lit/ds/symlink/drv8825.pdf
```

Comment:
Might make sense to collect these PDFs here in the repository, as to not be dependent on possibly fragile links. Probably not quite legal from a copyright perspective, but who's going to complain.

Again, just a note, no change required here.

Reply:
I did archive the URL, so I could change to the archive.org URL instead if you'd like. I think collecting PDFs in a folder is probably fine as well, I can't see anybody coming after us for that.

Reply:
Thanks, that's fine! I'll open an issue later to track this, no action required right now.